### PR TITLE
[routing] partly remove std/... from routing

### PR DESCRIPTION
**Pull request by gmoryes**

### CODE REVIEW

**Comment by gmoryes** [comment]
JTA

File: routing/index_router.hpp [hunk 1]
```diff
@@ -69,7 +68,7 @@ class IndexRouter : public IRouter
   IndexRouter(VehicleType vehicleType, bool loadAltitudes,
               CountryParentNameGetterFn const & countryParentNameGetterFn,
               TCountryFileFn const & countryFileFn, CourntryRectFn const & countryRectFn,
-              shared_ptr<NumMwmIds> numMwmIds, unique_ptr<m4::Tree<NumMwmId>> numMwmTree,
+              shared_ptr<NumMwmIds> numMwmIds, std::unique_ptr<m4::Tree<NumMwmId>> numMwmTree,
```
Comment (vmihaylenko) [inline routing/index_router.hpp]: std::shared_ptr

File: routing/joint_index.hpp [hunk 2]
```diff
@@ -52,7 +52,7 @@ class JointIndex final
     return m_offsets[nextId];
   }
 
-  vector<uint32_t> m_offsets;
-  vector<RoadPoint> m_points;
+  std::vector<uint32_t> m_offsets;
```
Comment (vmihaylenko) [inline routing/joint_index.hpp]: include <cstdint>

File: routing/online_absent_fetcher.hpp [hunk 3]
```diff
@@ -7,13 +7,13 @@
...
-#include "std/unique_ptr.hpp"
-#include "std/vector.hpp"
+#include <memory>
+#include <string>
+#include <vector>
 
 namespace routing
 {
-using TCountryLocalFileFn = function<bool(string const &)>;
+using TCountryLocalFileFn = std::function<bool(string const &)>;
```
Comment (vmihaylenko) [inline routing/online_absent_fetcher.hpp]: std::string

File: routing/features_road_graph.hpp [hunk 4]
```diff
@@ -48,7 +49,7 @@ class FeaturesRoadGraph : public IRoadGraph
     double const m_maxSpeed;
     double const m_offroadSpeedKMpH;
 
-    mutable map<MwmSet::MwmId, shared_ptr<VehicleModelInterface>> m_cache;
+    mutable std::map<MwmSet::MwmId, shared_ptr<VehicleModelInterface>> m_cache;
```
Comment (vmihaylenko) [inline routing/features_road_graph.hpp]: std::shared_ptr

File: routing/loaded_path_segment.hpp [hunk 5]
```diff
@@ -23,18 +23,18 @@ namespace routing
  */
 struct LoadedPathSegment
 {
-  vector<Junction> m_path;
-  vector<turns::SingleLaneInfo> m_lanes;
+  std::vector<Junction> m_path;
+  std::vector<turns::SingleLaneInfo> m_lanes;
   string m_name;
```
Comment (vmihaylenko) [inline routing/loaded_path_segment.hpp]: std::string

Comment (vmihaylenko) [reply routing/loaded_path_segment.hpp]: ```#include <string>```

File: routing/index_graph_serialization.hpp [hunk 6]
```diff
@@ -323,7 +323,7 @@ class IndexGraphSerializer final
     Joint::Id GetCount() const { return m_count; }
 
   private:
-    static uint32_t constexpr kEmptyEntry = numeric_limits<uint32_t>::max();
+    static uint32_t constexpr kEmptyEntry = std::numeric_limits<uint32_t>::max();
```
Comment (vmihaylenko) [inline routing/index_graph_serialization.hpp]: static auto constexpr? and below

File: routing/joint_index.hpp (hunk 2 above)
Comment (gmoryes) [reply routing/joint_index.hpp]: done

File: routing/online_absent_fetcher.hpp (hunk 3 above)
Comment (gmoryes) [reply routing/online_absent_fetcher.hpp]: done

File: routing/features_road_graph.hpp (hunk 4 above)
Comment (gmoryes) [reply routing/features_road_graph.hpp]: done

File: routing/loaded_path_segment.hpp (hunk 5 above)
Comment (gmoryes) [reply routing/loaded_path_segment.hpp]: done

File: routing/index_graph_serialization.hpp (hunk 6 above)
Comment (gmoryes) [reply routing/index_graph_serialization.hpp]: first yes, second no
I don't have to look recursively to know the type

**Comment by gmoryes** [comment]
JTA

File: routing/index_graph_serialization.hpp (hunk 6 above)
Comment (vmihaylenko) [reply routing/index_graph_serialization.hpp]: they're so close to each other...
could you please do ```numeric_limits<uint32_t>::max() - 1``` then?

Comment (gmoryes) [reply routing/index_graph_serialization.hpp]: done

**Comment by gmoryes** [comment]
JTA

**Comment by mapsmetest** [comment]
Analysed

**Comment by gmoryes** [comment]
JTA

**Comment by ToshUxanoff** [comment]
JTIOS

**Comment by ToshUxanoff** [comment]
JTIOS

**Pull request merged**